### PR TITLE
SLING-11844 Index definitions must not be extracted from empty nodes

### PR DESCRIPTION
**Pull request by abhishekgarg18**

### CODE REVIEW

**Review by rombert — CHANGES_REQUESTED** [review]
Thanks for the contribution @abhishekgarg18 ! I think it would be more straightforward to not emit the partial index definitions from `IndexDefinitionsEntryHandler` at all. Can you please update the PR to do that?

**Comment by abhishekgarg18** [comment]
I have the PR to filter out empty nodes to IndexDefinitions.java, However, if I have to move it here, I have to do special handling for txt files.

File: src/main/java/org/apache/sling/feature/cpconverter/index/IndexDefinitions.java [hunk 1]
```diff
@@ -74,6 +76,11 @@ public void addNode(@NotNull String parentPath, @NotNull DocViewNode2 node) {
         if ( existing != null ) {
             currentChildren.remove(existing);
         }
+        if ( CollectionUtils.isEmpty(node.getProperties())
+                && node.getName()!= null
```
Comment (rombert) [inline src/main/java/org/apache/sling/feature/cpconverter/index/IndexDefinitions.java:80]: SonarCloud says that you can remove this check, `getName()` never returns null

https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=169&id=apache_sling-org-apache-sling-feature-cpconverter&open=AYi4b0ONGxwo1pTO_aYE

File: src/main/java/org/apache/sling/feature/cpconverter/index/IndexDefinitions.java [hunk 2]
```diff
@@ -74,6 +76,11 @@ public void addNode(@NotNull String parentPath, @NotNull DocViewNode2 node) {
...
+                && !(node.getName().toString().contains(".txt")) ){
```
Comment (rombert) [inline src/main/java/org/apache/sling/feature/cpconverter/index/IndexDefinitions.java:81]: This check doesn't look right, I see the failing test in `IndexDefinitionsEntryHandlerTest#handleIndexDefinitionWithStopwordsInAnalyzer , but we should find a more generic condition besides "*.txt".

**Review by rombert — CHANGES_REQUESTED** [review]

**Pull request closed without merging**